ovirt_imageio/client/_options.py

```diff
@@ -10,62 +10,124 @@
 Tool options.
 """
 
+from collections import namedtuple
+
 from .. _internal.units import KiB, MiB, GiB, TiB
 
 import argparse
 import configparser
 import os
+import sys
+
+
+class Option(
+        namedtuple("Option", "name,args,config,required,type,default,help")):
+
+    __slots__ = ()
+
+    def __new__(cls, name=None, args=(), config=False, required=False,
+                type=str, default=None, help=None):
+        return tuple.__new__(
+            cls, (name, args, config, required, type, default, help))
+
+
+def log_level(s):
+    if s not in ("debug", "info", "warning", "error"):
+        raise ValueError(f"Invalid 'log_level' value: '{s}'")
+    return s
 
 
 class Parser:
 
+    _OPTIONS = [
+        Option(
+            name="config",
+            args=["-c", "--config"],
+            help="If set, read specified section from the configuration file.",
+        ),
+        Option(
+            name="engine_url",
+            args=["--engine-url"],
+            config=True,
+            required=True,
+            help=("ovirt-engine URL. If not set, read from the specified "
+                  "config section (required)."),
+        ),
+        Option(
+            name="username",
+            args=["--username"],
+            config=True,
+            required=True,
+            help=("ovirt-engine username. If not set, read from the "
+                  "specified config section (required)."),
+        ),
+        Option(
+            name="password",
+            config=True,
+        ),
+        Option(
+            name="password_file",
+            args=["--password-file"],
+            help=("Read ovirt-engine password from file. If not set, read "
+                  "password from the specified config section."),
+        ),
+        Option(
+            name="cafile",
+            args=["--cafile"],
+            config=True,
+            help=("Path to ovirt-engine CA certificate. If not set, read "
+                  "from the specified config section"),
+        ),
+        Option(
+            name="log_file",
+            args=["--log-file"],
+            config=True,
+            help="If set, log to file.",
+        ),
+        Option(
+            name="log_level",
+            args=["--log-level"],
+            config=True,
+            type=log_level,
+            default="info",
+            help="Logging level (default: info).",
```

Comment:
Should the levels be listed in the help? Some systems have more/different levels.

Reply:
Yes, we need to list the levels here.

Reply:
The choices are listed now.